src/core/io/eIO/io/einputfb.h

```diff
@@ -0,0 +1,151 @@
+ * Copyright (c) 2025 Maximilian Scharf
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Maximilian Scharf - Initial implementation.
+ *******************************************************************************/
+
+#pragma once
+
+#include "core/io/inputfb.h"
+#include "core/io/device/io_controller.h"
+#include "eGenAdapter_adp.h"
+#include "ESpec.h"
+#include "EConfig.h"
+#include "eConfigFB.h"
+
+USE_STRING_ID(eCONF);
+USE_STRING_ID(eGenAdapter);
+
+namespace forte::core::io {
+
+  template<class T>
+  class CeInputFB : public CInputFB<T> {
+    protected:
+      static const int scmeCONFAdpNum = 0;
+
+    private:
+      static const SAdapterInstanceDef scmAdapterInstances[];
+
+      using CInputFB<T>::scmEventInputNames;
+      using CInputFB<T>::scmEventInputTypeIds;
+      using CInputFB<T>::scmEIWith;
+      using CInputFB<T>::scmEIWithIndexes;
+
+      using CInputFB<T>::scmEventOutputNames;
+      using CInputFB<T>::scmEventOutputTypeIds;
+      using CInputFB<T>::scmEOWith;
+      using CInputFB<T>::scmEOWithIndexes;
+
+      using CInputFB<T>::scmDataInputNames;
+      using CInputFB<T>::scmDataInputTypeIds;
+      using CInputFB<T>::scmDataOutputNames;
+      using CInputFB<T>::scmDataOutputTypeIds;
+
+    protected:
+      using CInputFB<T>::var_IN;
+
+    public:
+      CeInputFB(forte::core::CFBContainer &paContainer, const CStringDictionary::TStringId paInstanceNameId)
+        : CInputFB<T>(paContainer, scmFBInterfaceSpec, paInstanceNameId),
+          var_eCONF(STRID(eCONF), *this, true) {};
+
+      FORTE_eGenAdapter var_eCONF;
+
+      EConfigFB *peerFB;
+
+      CAdapter *getAdapterUnchecked(const size_t paIndex) override {
+        switch(paIndex) {
+          case 0: return &var_eCONF;
+        }
+        return nullptr;
+      }
+
+      bool eventConfig() {
+        CAdapter* peerAdapter = nullptr;
+        FORTE_eGenAdapter* eGenAdapter = nullptr;
+
+        if ((peerAdapter = getAdapterUnchecked(scmeCONFAdpNum)->getPeer()) != nullptr){
+          eGenAdapter = static_cast<FORTE_eGenAdapter*>(peerAdapter);
+          peerFB = static_cast<EConfigFB *>(eGenAdapter->parentFB);
+          peerFB->eventGen();
+        }
+        return true;
+      }
+
+    protected:
+      static const SFBInterfaceSpec scmFBInterfaceSpec;
+      std::forward_list<ESpecBase*> *eventGenList;
+
+      virtual bool checkEventTriggers() {
+        return false;
+      };
+
+    private:
+      void executeEvent(TEventID paEIID, CEventChainExecutionThread *const paECET) final override {
+        switch(paEIID){
+          case cgExternalEventID:
+            if(this->var_QI) {
+              this->var_QO = this->read();
+              if (this->var_QO && peerFB && peerFB->checkEventTriggersConf(&var_IN)) {
```

Comment:
this should be moved to on change so that you are only generating an external event when the even ttrigger says so.